snap/snapcraft.yaml

```diff
@@ -0,0 +1,47 @@
+name: brogue-ce
+summary: Welcome to the Dungeons of Doom!
+description: Brogue is a single-player strategy game set in the halls of a mysterious and randomly-generated dungeon.
+license: AGPL-3.0
+
+adopt-info: brogue-ce
+confinement: strict
+base: core18
+
+parts:
+  brogue-ce:
+    source: .
+    plugin: make
+
+    override-pull: |
+      snapcraftctl pull
+      version="$(git describe --always | sed -e 's/-/+git/;y/-/./')"
+      [ -n "$(echo $version | grep "+git")" ] && grade=devel || grade=stable
+      snapcraftctl set-version "$version"
+      snapcraftctl set-grade "$grade"
+
+    override-build: |
+      # Build and install brogue
+      make clean && make DATADIR='\$$SNAP/usr/share' && make install DESTDIR=$SNAPCRAFT_PART_INSTALL &&
```

Comment:
Is there no way to hard-code snap directories? Do all Snap applications have to dynamically expand this environment variable in the same way?

Reply:
AFAIK, no, the snap gets installed into a directory containing the install order:

```
pbovbel@pbovbel-dell:/var/snap/brogue-ce$ ll /var/snap/brogue-ce/
total 20
drwxr-xr-x  5 root root 4096 May 21 18:36 ./
drwxr-xr-x 25 root root 4096 May 21 16:03 ../
drwxr-xr-x  2 root root 4096 May 21 16:03 common/
lrwxrwxrwx  1 root root    3 May 21 18:36 current -> x28/
drwxr-xr-x  2 root root 4096 May 21 16:03 x27/
drwxr-xr-x  2 root root 4096 May 21 16:03 x28/
```

I suppose it was a toss-up between implementing environment expansion for datadir, or just allowing it to be passed in as an argument to `brogue`. I went with the former.